CMakeLists.txt

```diff
@@ -3,9 +3,52 @@
 # See LICENSE.txt for the full license.
 
 cmake_minimum_required(VERSION 3.0)
+
+include("cmake/HunterGate.cmake")
+HunterGate(
+    URL "https://github.com/ruslo/hunter/archive/v0.16.15.tar.gz"
+    SHA1 "6974c2150fc0d3b09de3ad1efcbf15d360647ffa"
+)
+
 project(dlib VERSION 19.0)
 
-add_library(dlib INTERFACE)
+if(DLIB_HEADER_ONLY)
+  add_library(dlib INTERFACE)
+else()
+  
+  #duplicating dlib's default options config
```

Comment:
I don't think that it's correct. For example `DLIB_ISO_CPP_ONLY` is an option that set in `dlib/CMakeLists.txt` obviously before top-level `CMakeLists.txt`.

> need to alter the defaults before they are created

What for? If you need configuration that differs from default then you have to explicitly set your personal options by `-D` to CMake or by `CMAKE_ARGS` in Hunter.